barretenberg/cpp/src/barretenberg/vm2/tracegen/tx_trace.cpp

```diff
@@ -605,6 +608,9 @@ void TxTraceBuilder::process(const simulation::EventEmitterInterface<simulation:
 
         // We have events to process in this phase
         for (const auto& tx_phase_event : phase_events) {
+            // If this phase is the first revert, set tx_reverted:
+            tx_reverted |= tx_phase_event->reverted;
```

Comment:
This works but I think it's formally speaking a bitwise OR. C++ doesn't have `||=` which is the boolean OR. To be safe it might be better to do `tx_reverted = tx_reverted || tx_phase_event->reverted;`

In this case it doesn't matter much but for example, `|=` does not do short=circuit and the above expression would.